bycycle/burst/utils.py

```diff
@@ -49,3 +49,73 @@ def check_min_burst_cycles(is_burst, min_n_cycles=3):
             temp_cycle_count = 0
 
     return is_burst
+
+
+def recompute_edges(df_features, threshold_kwargs):
+    """Recompute the is_burst column for cycles on the edges of bursts.
+
+    Parameters
+    ----------
+    df_features : pandas.DataFrame
+        A dataframe containing shape and burst features for each cycle.
+    threshold_kwargs : dict, optional, default: None
+        Feature thresholds for cycles to be considered bursts, matching keyword arguments for:
+
+        - :func:`~.detect_bursts_cycles` for consistency burst detection
+          (i.e. when burst_method == 'cycles')
+
+    Returns
+    -------
+    df_features_edges : pandas.DataFrame
+        An cycle feature dataframe with an updated ``is_burst`` column for edge cycles.
+
+    Notes
+    -----
+
+    - `df_features` must be computed using consistency burst detection.
+
+    Examples
+    --------
+    Lower the amplitude consistency threshold to zero for cycles on the edges of bursts:
+
+    >>> sig = sim_combined(n_seconds=4, fs=1000, components={'sim_bursty_oscillation': {'freq': 10},
+    ...                                                      'sim_powerlaw': {'exp': 2}})
+    >>> threshold_kwargs = {'amp_fraction_threshold': 0., 'amp_consistency_threshold': .5,
+    ...                     'period_consistency_threshold': .5, 'monotonicity_threshold': .4,
+    ...                     'min_n_cycles': 3}
+    >>> df_features, _ = compute_features(sig, fs=1000, f_range=(8, 12), threshold_kwargs=threshold_kwargs)
+    >>> threshold_kwargs['amp_consistency_threshold'] = 0
+    >>> df_features_edges = recompute_edges(df_features, threshold_kwargs)
+    """
+
+    from bycycle.burst import detect_bursts_cycles
```

Comment:
i presume this is imported within the function to avoid a circular import? Perhaps worth adding a quick comment noting that's why.